### PR TITLE
John conroy/storybook

### LINKED ISSUE #1884: Create component library for shared ui components
We've decided to create a component library to make shared components more accessible and promote better practices.

To start we need to:

- [ ] Create git submodule
- [ ] Host the component library on github pages or similar

### PR DESCRIPTION
**Pull request by john-conroy**
Fix #1884. Configures storybook and adds a very basic first component. I'm working on #1909 as a more in-depth story example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,8 +70,8 @@ These are concatenated by `push.sh`.
 <details><summary>For React</summary>
 
 - Components with tests or styles should be placed in to their own directory.
-- Styles should be placed in `style.*` where the extension is js for styled components or css for stylesheets.
-- Tests should be placed in `*.spec.js` where the prefix is the name of the component.
+- Styles should follow the `style.*` pattern where the extension is js for styled components or css for stylesheets.
+- Tests and stories should follow the `*.spec.js` and `*.stories.js` patterns respectively where the prefix is the name of the component.
```
Comment (mccalluc) [inline README.md:74]:
```suggestion
- Tests should follow the `*.spec.js` pattern...
- and stories should follow the `*.stories.js` pattern. For both, the prefix is the name of the component.
```

File: README.md [hunk 2]
```diff
@@ -104,6 +104,9 @@ npm run lint
 npm run lint:fix
 ```
 
+### Storybook
+To start storybook locally run `npm run storybook`.
```
Comment (mccalluc) [inline README.md:108]: Would it be useful to just wrap this into `dev-start.sh`?

File: context/app/static/js/shared-styles/alerts/OutlinedAlert.stories.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  },
+};
+
+const Template = (args) => <Alert {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
+  severity: 'warning',
+  children: 'Alert, intruders!',
+};
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/alerts/OutlinedAlert.stories.js:22]: Will an epilogue like this be standard for us? Just skimming the doc, should we be trying to follow naming conventions so [single story hoisting](https://storybook.js.org/docs/riot/writing-stories/naming-components-and-hierarchy#single-story-hoisting) applies?

File: context/build-utils/aliases.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+const aliases = {
+  'metadata-field-descriptions$': resolve('./ingest-validation-tools/docs/field-descriptions.yaml'),
+  js: resolve(__dirname, '../app/static/js/'),
+  'portal-images': resolve(__dirname, '../app/static/portal-images/'),
+  'shared-styles': resolve(__dirname, '../app/static/js/shared-styles/'),
+  package: resolve(__dirname, '../package.json'),
+};
+
+exports.aliases = aliases;
```
Comment (mccalluc) [inline context/build-utils/aliases.js:12]:
```suggestion
const alias = {
  'metadata-field-descriptions$': resolve('./ingest-validation-tools/docs/field-descriptions.yaml'),
  js: resolve(__dirname, '../app/static/js/'),
  'portal-images': resolve(__dirname, '../app/static/portal-images/'),
  'shared-styles': resolve(__dirname, '../app/static/js/shared-styles/'),
  package: resolve(__dirname, '../package.json'),
};

exports.alias = alias;
```
and corresponding changes in other files? I agree that the plural makes more sense... but there seems to be a convention, and probably easier just to follow it?

**Review by mccalluc — APPROVED** [review]
A couple things to consider, but I think it could be merged as is, if you like the status quo. I'll need to learn more about the best idioms for writing stories.

File: README.md (hunk 2 above)
Comment (john-conroy) [reply README.md]: Added to `dev-start.sh`.

File: context/app/static/js/shared-styles/alerts/OutlinedAlert.stories.js (hunk 3 above)
Comment (john-conroy) [reply context/app/static/js/shared-styles/alerts/OutlinedAlert.stories.js]: Fixed relevant stories.

File: context/build-utils/aliases.js (hunk 4 above)
Comment (john-conroy) [reply context/build-utils/aliases.js]: Changed, thanks!

File: README.md [hunk 5]
```diff
@@ -104,6 +105,11 @@ npm run lint
 npm run lint:fix
 ```
 
+### Storybook
+To start storybook locally you can run `npm run storybook` or visit [localhost:6006](http://localhost:6006). after you have run `./dev-start.sh`.
```
Comment (mccalluc) [inline README.md:109]:
```suggestion
To start storybook locally you can either run `./dev-start.sh`, or just `npm run storybook`,
and after it has started, visit [localhost:6006](http://localhost:6006).
```

**Review by mccalluc — APPROVED** [review]
wording tweak.

**Pull request merged**